runtime-sdk/src/modules/consensus/mod.rs

```diff
@@ -0,0 +1,279 @@
+    ) -> Result<(), Error> {
+        Self::ensure_consensus_denomination(ctx, amount.denomination())?;
+
+        ctx.emit_message(
+            Message::Staking {
+                v: 0,
+                msg: StakingMessage::Withdraw(staking::Withdraw {
+                    from: from.into(),
+                    amount: amount.amount().clone(),
+                }),
+            },
+            hook,
+        )?;
+
+        Ok(())
+    }
+
+    fn escrow<C: Context>(
+        ctx: &mut C,
+        to: Address,
+        amount: &token::BaseUnits,
+        hook: MessageEventHookInvocation,
+    ) -> Result<(), Error> {
+        Self::ensure_consensus_denomination(ctx, amount.denomination())?;
+
+        ctx.emit_message(
+            Message::Staking {
+                v: 0,
+                msg: StakingMessage::AddEscrow(staking::Escrow {
+                    account: to.into(),
+                    amount: amount.amount().clone(),
+                }),
+            },
+            hook,
+        )?;
+
+        Ok(())
+    }
+
+    fn reclaim_escrow<C: Context>(
+        ctx: &mut C,
+        from: Address,
+        amount: &token::BaseUnits,
+        hook: MessageEventHookInvocation,
+    ) -> Result<(), Error> {
+        Self::ensure_consensus_denomination(ctx, amount.denomination())?;
+
+        ctx.emit_message(
+            Message::Staking {
+                v: 0,
+                msg: StakingMessage::ReclaimEscrow(staking::ReclaimEscrow {
+                    account: from.into(),
+                    shares: amount.amount().clone(),
+                }),
+            },
+            hook,
+        )?;
+
+        Ok(())
+    }
+
+    fn consensus_denomination<C: Context>(ctx: &mut C) -> Result<token::Denomination, Error> {
+        let params = Self::params(ctx.runtime_state());
+        Ok(params.consensus_denomination)
+    }
+
+    fn account<C: Context>(ctx: &C, addr: Address) -> Result<ConsensusAccount, Error> {
+        let state = StakingImmutableState::new(ctx.consensus_state());
+        state
+            .account(ctx.io_ctx(), addr.into())
+            .map_err(Error::InternalStateError)
+    }
+}
+
+impl module::Module for Module {
+    const NAME: &'static str = MODULE_NAME;
+    const VERSION: u32 = 1;
+    type Error = Error;
+    type Event = Event;
+    type Parameters = Parameters;
+}
+
+impl module::MethodRegistrationHandler for Module {}
+impl module::MessageHookRegistrationHandler for Module {}
+
+impl module::MigrationHandler for Module {
+    type Genesis = Genesis;
+
+    fn init_or_migrate(
+        ctx: &mut DispatchContext<'_>,
+        meta: &mut modules::core::types::Metadata,
+        genesis: &Self::Genesis,
+    ) -> bool {
+        let version = meta.versions.get(Self::NAME).copied().unwrap_or_default();
+        if version == 0 {
+            // TODO: enable loading consensus denomination from consensus state after:
+            // https://github.com/oasisprotocol/oasis-core/issues/3868
```

Comment:
Consider opening an issue in the SDK repository as well.

Reply:
Done https://github.com/oasisprotocol/oasis-sdk/issues/111